### PR TITLE
Fix bug in `sc.bin` when rebinning to multi-dim coord

### LINKED ISSUE #2354: `sc.bin` wrong when rebinning to multi-dimensional coord
```python
import scipp as sc

da = sc.data.binned_xy(100, 2, 1)
del da.coords['x']
y = sc.array(dims=['x', 'y'], unit='m', values=[[0.0, 0.5, 1.0], [0.0, 0.9, 1.0]])
sc.bin(da, edges=[y]).bins.size().values
```
gives something like
```
array([[25, 29],
       [22, 24]])
```

If we first `del da.coords['y']` and then call `bin`, we get the correct result:
```
array([[25, 29],
       [41,  5]])
```

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #2354.

### CODE REVIEW

File: lib/dataset/test/bin_test.cpp [hunk 1]
```diff
@@ -226,6 +226,22 @@ TEST_P(BinTest, 2d_drop_out_of_group) {
...
+TEST_P(
+    BinTest,
+    rebin_inner_1d_coord_to_2d_coord_gives_same_result_as_direct_binning_to_2d_coord) {
+  auto table = GetParam();
+  auto xy = bin(table, {edges_x_coarse, edges_y_coarse});
+  Variable edges_y_2d = makeVariable<double>(Dims{Dim::X, Dim::Y}, Shape{2, 3},
+                                             Values{-2, 1, 2, -3, 0, 3});
+  // With bin-coord for Y
+  expect_near(bin(xy, {edges_y_2d}),
+              bin(bin(table, {edges_x_coarse}), {edges_y_2d}));
```
Comment (jl-wynen) [inline lib/dataset/test/bin_test.cpp:238]: Why bin in `edges_x_coarse`? Is this not a noop?

**Review by jl-wynen — APPROVED** [review]
Looks good

File: lib/dataset/test/bin_test.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply lib/dataset/test/bin_test.cpp:238]: `table` is not binned at all yet, i.e., does not depend on `x`. To be able to bin by 2d `y`-edges, we first need to bin in `x`.

Comment (jl-wynen) [reply lib/dataset/test/bin_test.cpp:238]: Ah, right. I confused it with `xy`

**Pull request merged**